SuccessBonusActivationActivity/handler.ts

```diff
@@ -0,0 +1,181 @@
+import {
+  NewUserBonus,
+  RetrievedUserBonus,
+  UserBonusModel
+} from "../models/user_bonus";
+
+export const SuccessBonusActivationInput = t.interface({
+  bonusActivation: BonusActivationWithFamilyUID
+});
+
+export const SuccessBonusActivationSuccess = t.interface({
+  kind: t.literal("SUCCESS")
+});
+export type SuccessBonusActivationSuccess = t.TypeOf<
+  typeof SuccessBonusActivationSuccess
+>;
+
+export const InvalidInputFailure = t.interface({
+  kind: t.literal("INVALID_INPUT")
+});
+export type InvalidInputFailure = t.TypeOf<typeof InvalidInputFailure>;
+
+export const UnhandledFailure = t.interface({
+  kind: t.literal("UNHANDLED_FAILURE"),
+  reason: t.string
+});
+export type UnhandledFailure = t.TypeOf<typeof UnhandledFailure>;
+
+export const TransientFailure = t.interface({
+  kind: t.literal("TRANSIENT")
+});
+export type TransientFailure = t.TypeOf<typeof TransientFailure>;
+
+const SuccessBonusActivationFailure = t.union(
+  [InvalidInputFailure, UnhandledFailure, TransientFailure],
+  "SuccessBonusActivationFailure"
+);
+export type SuccessBonusActivationFailure = t.TypeOf<
+  typeof SuccessBonusActivationFailure
+>;
+
+export const SuccessBonusActivationResult = t.taggedUnion("kind", [
+  SuccessBonusActivationSuccess,
+  SuccessBonusActivationFailure
+]);
+export type SuccessBonusActivationResult = t.TypeOf<
+  typeof SuccessBonusActivationResult
+>;
+
+/**
+ * Converts a Promise<Either> into a TaskEither
+ * This is needed because our models return unconvenient type. Both left and rejection cases are handled as a TaskEither left
+ * @param lazyPromise a lazy promise to convert
+ *
+ * @returns either the query result or a query failure
+ */
+const fromQueryEither = <R>(
+  lazyPromise: () => Promise<Either<QueryError | Error, R>>
+): TaskEither<Error, R> =>
+  tryCatch(lazyPromise, toError).chain(errorOrResult =>
+    fromEither(errorOrResult).mapLeft(toError)
+  );
+
+const updateBonusAsActive = (
+  bonusActivationModel: BonusActivationModel,
+  bonusActivation: BonusActivationWithFamilyUID
+): TaskEither<Error, RetrievedBonusActivation> => {
+  return fromQueryEither(() => {
+    const bonusToUpdate: BonusActivationWithFamilyUID = {
+      ...bonusActivation,
+      status: BonusActivationStatusEnum.ACTIVE
+    };
+    return bonusActivationModel.replace(bonusToUpdate);
+  });
+};
+
+const saveBonusForEachFamilyMember = (
+  userBonusModel: UserBonusModel,
+  {
+    applicantFiscalCode,
+    id: bonusId,
+    dsuRequest: { familyMembers }
+  }: BonusActivationWithFamilyUID
+): TaskEither<Error, readonly RetrievedUserBonus[]> =>
+  array.sequence(taskEither)(
+    familyMembers
+      .map<NewUserBonus>(({ fiscalCode }) => ({
+        bonusId,
+        fiscalCode,
+        id: (bonusId as unknown) as NonEmptyString,
+        isApplicant: applicantFiscalCode === fiscalCode,
+        kind: "INewUserBonus"
+      }))
+      .map(nub => () => userBonusModel.create(nub, bonusId))
+      .map(fn => fromQueryEither(fn))
```

Comment:
We can use more readable variables name?
```suggestion
      .map(newUserBonus =>
        fromQueryEither(() => userBonusModel.create(newUserBonus, bonusId))
      )
```